launch/src/read.rs

```diff
@@ -35,44 +35,85 @@
 // www.navitia.io
 
 use super::config;
-use crate::{config::LaunchParams, loki::TransitData};
+use crate::config::launch_params::LocalFileParams;
+use crate::loki::TransitData;
 use anyhow::{format_err, Error};
 use loki::{
     models::base_model::{self, BaseModel},
     tracing::{info, warn},
-    transit_model, DataIO, DataTrait, LoadsData,
+    transit_model, DataIO, DataTrait, LoadsData, PositiveDuration,
 };
-use std::{collections::BTreeMap, str::FromStr, time::SystemTime};
+use std::path::PathBuf;
+use std::{str::FromStr, time::SystemTime};
 
 pub fn read(launch_params: &config::LaunchParams) -> Result<(TransitData, BaseModel), Error> {
-    let base_model = read_model(launch_params)?;
+    let base_model = read_model(
+        &LocalFileParams {
+            input_data_path: launch_params.input_data_path.clone(),
+            loads_data_path: launch_params.loads_data_path.clone(),
+        },
+        launch_params.input_data_type.clone(),
+        launch_params.default_transfer_duration,
+    )?;
 
     let data = build_transit_data(&base_model);
 
     Ok((data, base_model))
 }
 
-pub fn read_model(launch_params: &LaunchParams) -> Result<BaseModel, Error> {
-    let model = match launch_params.input_data_type {
-        config::InputDataType::Ntfs => transit_model::ntfs::read(&launch_params.input_data_path)?,
+pub fn read_model_from_reader<R>(
+    input_data_reader: R,
```

Comment:
```suggestion
pub fn read_model_from_zip_reader<R>(
    input_data_zip_reader: R,
```
since this function expect a reader on a zip file